src/org/apache/cocoon/transformation/GitTransformer.java

```diff
@@ -0,0 +1,737 @@
+    private static final String FILE_ELEMENT = "file";
+    private static final String OBJECT_ID_ATTR = "object-id";
+    private static final String NAME_ATTR = "name";
+    private static final String REMOTE_ATTR = "remote";
+    private static final String REMOTE_BRANCH_ATTR = "remote-branch";
+    private static final String FETCHED_FROM_ATTR = "fetched-from";
+    private static final String FETCH_MESSAGES_ATTR = "fetch-messages";
+    private static final String MERGE_STATUS_ATTR = "merge-status";
+    private static final String CHANGE_TYPE_ATTR = "change-type";
+    private static final String SCORE_ATTR = "score";
+    private static final String NEW_PATH_ATTR = "new-path";
+    private static final String OLD_PATH_ATTR = "old-path";
+    private static final String NEW_ID_ATTR = "new-id";
+    private static final String OLD_ID_ATTR = "old-id";
+    private static final String NEW_MODE_ATTR = "new-mode";
+    private static final String OLD_MODE_ATTR = "old-mode";
+    private static final String ADDED_ELEMENT = "added";
+    private static final String CHANGED_ELEMENT = "changed";
+    private static final String CONFLICTING_ELEMENT = "conflicting";
+    private static final String IGNOREDNOTININDEX_ELEMENT = "ignorednotinindex";
+    private static final String MISSING_ELEMENT = "missing";
+    private static final String MODIFIED_ELEMENT = "modified";
+    private static final String REMOVED_ELEMENT = "removed";
+    private static final String UNTRACKED_ELEMENT = "untracked";
+    private static final String UNTRACKEDFOLDERS_ELEMENT = "untrackedfolders";
+
+    private String commit_message;
+    private String repository;
+    private String author_name;
+    private String author_email;
+
+    public GitTransformer() {
+        this.defaultNamespaceURI = GIT_NAMESPACE_URI;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.apache.avalon.framework.configuration.Configurable#configure(org.
+     * apache.avalon.framework.configuration.Configuration)
+     */
+    @Override
+    public void configure(Configuration configuration)
+            throws ConfigurationException {
+        super.configure(configuration);
+    }
+
+    @Override
+    public void setup(SourceResolver resolver, Map objectModel, String src,
+            Parameters params) throws ProcessingException, SAXException, IOException {
+        super.setup(resolver, objectModel, src, params);
+        reset();
+    }
+
+    private String getAttribute(Attributes attr, String name, String defaultValue) {
+        return (attr.getIndex(name) >= 0) ? attr.getValue(name) : defaultValue;
+    }
+
+    public void startTransformingElement(String uri, String name, String raw, Attributes attr)
+            throws ProcessingException, IOException, SAXException {
+        if (uri.equals(GIT_NAMESPACE_URI)) {
+            
+            if (name.equals(CLONE_ELEMENT)) {
+                final String repository = getAttribute(attr, REPOSITORY_ATTR, null);
```

Comment:
Er is een (1) `if` en de `else` die niet het repository attribuut ophalen en checken. Dat zijn 4 regels code per vertakking die je 11 keer herhaalt (en dan tel ik niet de lege regels). Dat kan handiger en duidelijker! Bijvoorbeeld door:
```
    private String getAttribute(Attributes attr, String name) throws SAXException {
        if (attr.getIndex(name) >= 0) {
          return attr.getValue(name);
        } else {
          throw new SAXException(java.lang.String.format("Missing @%s attribute.", name));
        }
    }
```
En dan kun je zelfs dit doen:
```
doClone(
  getAttribute(attr, REPOSITORY_ATTR),
  getAttribute(attr, ACCOUNT_ATTR, null),
  getAttribute(attr, PASSWORD_ATTR, null),
  getAttribute(attr, URL_ATTR),
  getAttribute(attr, BRANCH_ATTR, MASTER_BRANCH)
);
```
En iets dergelijks voor de andere vertakkingen van de if.

Een lange geneste if-else wordt meestal geschreven als switch-statement.